src/utils/Hooks/Calculations/useAppropriateChartData.ts

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useRef, useState } from 'react';
+import { differenceInMonths, differenceInWeeks } from 'date-fns';
+import { CategoryCodes, ChartData, MeasurementTypeCodesLabel, TimeUnitCodes } from '../../../types/chartDataTypes';
+
+export const useAppropriateChartData = (
+    chartDataForGender: ChartData,
+    dateOfBirth: Date,
+    defaultIndicator: string,
+    gender: string,
+    setDefaultIndicatorError: (value: boolean) => void,
+) => {
+    const [selectedCategory, setSelectedCategory] = useState<keyof typeof CategoryCodes>();
+    const [selectedDataset, setSelectedDataset] = useState<string>();
+
+    const selectDatasetForCategoryRef = useRef<(category: keyof typeof CategoryCodes) => void>();
+    selectDatasetForCategoryRef.current = (category: keyof typeof CategoryCodes) => {
+        const { datasets } = chartDataForGender[category];
+        const childAgeInWeeks = differenceInWeeks(new Date(), dateOfBirth);
+        const childAgeInMonths = differenceInMonths(new Date(), dateOfBirth);
+        Object.entries(datasets)
+            .some(([key, value]) => {
+                const { range } = value.metadata;
+                const xAxis = value.metadata.xAxisLabel;
+
+                if (Object.values(MeasurementTypeCodesLabel)
+                    .includes(xAxis)) {
+                    setSelectedDataset((prevDataset) => (prevDataset !== key ? key : prevDataset));
+                    return true;
+                }
+
+                if (xAxis === TimeUnitCodes.weeks && childAgeInWeeks <= 13) {
+                    setSelectedDataset((prevDataset) => (prevDataset !== key ? key : prevDataset));
+                    return true;
+                }
+
+                const isChildAgeInRange = xAxis === TimeUnitCodes.months && childAgeInMonths >= range.start && childAgeInMonths <= range.end;
```

Comment:
Small comment as the previous, should be >= on start as it it, but < at the end. If a child is 24.5 (about 24 months and 2 weeks) months old, the nurse is directed to the 0 - 24 months, when the plot is plotted in 24 - 60.

```suggestion
                const isChildAgeInRange = xAxis === TimeUnitCodes.months && childAgeInMonths >= range.start && childAgeInMonths < range.end;
```